docs/source/name_inference.rst

```diff
@@ -0,0 +1,428 @@
+.. currentmodule:: torch
+
+.. _name_inference_reference-doc:
+
+Named Tensors operator coverage
+===============================
+
+Please read :ref:`named_tensors-doc` first for an introduction to named tensors.
+
+This document is a reference for *name inference*, a process for how named tensors
```

Comment:
I think numbering here would make things clearer.  Something like..."a process that defines how named tensors:
1.  use names to provide additional automatic runtime correctness checks
2.  propagate names from input tensors to output tensors
"
(for symmetry you want to say either "from inputs to outputs" or "from input tensors to output tensors".